### PR TITLE
multi az support

### PR DESCRIPTION
**Pull request by wallies**

### CODE REVIEW

**Comment by wallies** [comment]
this should solve #66

File: roles/consul/defaults/main.yml [hunk 1]
```diff
@@ -5,7 +5,8 @@ consul_dc: dc1
 consul_servers_group: consul_servers
 consul_advertise: "{{ ansible_default_ipv4.address }}"
 consul_bind_addr: "{{ ansible_default_ipv4.address }}"
-consul_bootstrap_expect: "{{ groups[consul_servers_group] | length }}"
+consul_retry_join: "{% for host in groups[consul_servers_group] | intersect(groups[consul_dc]) %}\"{{ hostvars[host].ansible_default_ipv4.address }}\"{% if not loop.last %}, {% endif %}{% endfor %}"
```
Comment (tayzlor) [inline roles/consul/defaults/main.yml]: whats the significance of adding the retry join?

File: terraform/aws/elb.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    healthy_threshold = 2
+    unhealthy_threshold = 2
+    timeout = 3
+    target = "HTTP:80/"
+    interval = 30
+  }
+
+  instances = ["${aws_instance.mesos-slave.*.id}"]
+  cross_zone_load_balancing = true
+}
```
Comment (tayzlor) [inline terraform/aws/elb.tf]: newline EOF needed

File: terraform/aws/variables.tf [hunk 3]
```diff
@@ -43,9 +43,20 @@ variable "public_subnet_cidr_block" {
   default     = "10.0.0.0/24"
 }
 
-variable "private_subnet_cidr_block" {
-  description = "Cidr block for private Mesos subnet."
-  default = "10.0.1.0/24"
+variable "zones" {
+    default = {
+        zone0 = "eu-west-1a"
```
Comment (tayzlor) [inline terraform/aws/variables.tf]: for styling / consistency might be worth either keeping all of these like ```zone0``` or moving to ```zone-0``` like the master_ips var. Otherwise move the master_ips to be like ```master1```

**Comment by tayzlor** [comment]
Initial overview of the code looks like it is good. Can we also update the architecture diagram on the frontpage with this change? 

To do that you need to provision the cluster as normal, then login to visualops then there is an option to import from AWS (and you can import the running instances) which should create a diagram which you can view/edit

File: roles/consul/defaults/main.yml (hunk 1 above)
Comment (wallies) [inline roles/consul/defaults/main.yml]: consul was failing to elect a leader and then failing to show up as services in the UI. Adding this fixed it.
-retry-join - Similar to -join but allows retrying a join if the first attempt fails. This is useful for cases where we know the address will become available eventually.

Comment (pmbauer) [inline roles/consul/defaults/main.yml]: fwiw, I've seen consul consistently fail to cluster on `bootstrap/apollo-launch.sh` and every consul agent ends up with a cluster of one member, itself.  We worked around it by defining `consul_join` but this is more flexible.

Comment (tayzlor) [inline roles/consul/defaults/main.yml]: @pmbauer I've seen that too occassionally - I reported it over in https://github.com/Capgemini/Apollo/pull/198. Looks like the consul related fixes here might address that too

Comment (tayzlor) [inline roles/consul/defaults/main.yml]: can we separate the consul retry join to a separate PR as that may resolve issues over in #198 and to avoid blocking on waiting for multi-az

**Comment by wallies** [comment]
couldnt add variables, as cant have dots in bash variables
export TF_VAR_zones.zone-0=${TF_VAR_zones.zone-0:-eu-west-1a}
export TF_VAR_zones.zone-1=${TF_VAR_zones.zone-1:-eu-west-1b}
export TF_VAR_zones.zone-2=${TF_VAR_zones.zone-2:-eu-west-1c}

**Comment by enxebre** [comment]
Closing in favour of https://github.com/Capgemini/Apollo/pull/385
